File: EPI/12.4/swift/12.4_test.swift
import Darwin

let tests: [(array: [Int], index: Int)] = [
    (
        array: [0],
        index: 0
    ),
    (
        array: [0, 1, 2, 3, 4],
        index: 0
    ),
    (
        array: [1, 2, 3, 4, 0],
        index: 4
    ),
    (
        array: [3, 4, 0, 1, 2],
        index: 2
    ),
]

for test in tests {
    let index = indexOfSmallest(test.array)
    guard index == test.index else {
        print("For cyclically sorted test array \(test.array), " +
            "expected index of smallest element to be \(test.index), but was \(index)")
        exit(1)
    }
}

File: EPI/12.4/typescript/12.4_test.ts
import indexOfMin from './12.4';

declare function require(name: string): any;
const assert = require('assert');
const inspect = (v: any) => require('util').inspect(v, {depth: null});

const tests: {array: number[], indexOfMin: number}[] = [
    {
        array: [0],
        indexOfMin: 0
    },
    {
        array: [0, 1, 2, 3, 4],
        indexOfMin: 0
    },
    {
        array: [1, 2, 3, 4, 0],
        indexOfMin: 4
    },
    {
        array: [3, 4, 0, 1, 2],
        indexOfMin: 2
    },
];

for (const test of tests) {
    const actual = indexOfMin(test.array);
    assert.strictEqual(actual, test.indexOfMin, `For array ${inspect(test.array)}, ` +
        `expected indexOfMin to be ${test.indexOfMin}, but was ${actual}`);
}
